rio_tiler/io/cogeo.py

```diff
@@ -346,6 +348,24 @@ def tile(
             # Buffered Tile Size
             tilesize += int(tile_buffer * 2)
 
+        if aoi is not None:
+            cutline = create_cutline(self.dataset, aoi, geometry_crs="epsg:4326")
+            vrt_options = kwargs.pop("vrt_options", {})
+            vrt_options.update({"cutline": cutline})
+
+            return self.part(
+                tile_bounds,
+                dst_crs=self.tms.rasterio_crs,
+                bounds_crs=None,
+                height=tilesize,
+                width=tilesize,
+                max_size=None,
+                indexes=indexes,
+                expression=expression,
+                vrt_options=vrt_options,
+                **kwargs,
+            )
+
```

Comment:
I don't like that `aio` forces you to repeat the 10-line return statement verbatim with the only change being for `vrt_options`. How about:
```suggestion
        vrt_options = None
        if aoi is not None:
            cutline = create_cutline(self.dataset, aoi, geometry_crs="epsg:4326")
            vrt_options = kwargs.pop("vrt_options", {})
            vrt_options.update({"cutline": cutline})

```
Then just passing `vrt_options=vrt_options` in the single return

Reply:
Or better:
```
vrt_options = kwargs.pop("vrt_options", {})
if aoi is not None:
    cutline = ...
    vrt_options.update(...)

return self.part(
    ...
    vrt_options=vrt_options
    ...
)
```

Reply:
Thanks for that. That's exactly what I was looking for 😄  e670c1e05f249493adc4144f291364fa64f14cc0